backend/tilavarauspalvelu/models/payment_order/actions.py

```diff
@@ -109,3 +132,72 @@ def update_order_status(self, new_status: OrderStatus, payment_id: str = "") ->
 
         EmailService.send_reservation_confirmed_email(reservation=reservation)
         EmailService.send_reservation_confirmed_staff_notification_email(reservation=reservation)
+
+    def cancel_together_with_verkkokauppa(self, *, cancel_on_error: bool = False) -> None:
+        if self.payment_order.remote_id is None or settings.MOCK_VERKKOKAUPPA_API_ENABLED:
+            if self.payment_order.status in OrderStatus.can_be_cancelled_statuses:
```

Comment:
Since this can be executed through a task, I guess `can_be_cancelled_statuses` should generally be the first thing checked?

Reply:
Fixed